src/GovUk.Education.ExploreEducationStatistics.Admin/Services/ReleaseService.cs

```diff
@@ -101,6 +101,30 @@ public class ReleaseService : IReleaseService
                 .OnSuccess(release => _mapper.Map<ReleaseViewModel>(release));
         }
 
+        public async Task<Either<ActionResult, List<ReleaseInternalNoteViewModel>>> GetInternalReleaseNotes(
+            Guid releaseId)
+        {
+            return await _persistenceHelper
+                .CheckEntityExists<Release>(releaseId)
+                .OnSuccess(_userService.CheckCanViewRelease)
+                .OnSuccess(_ =>
+                    _context.ReleaseStatus
```

Comment:
You can avoid this seperate fetch of all the release statuses filtered by release id by including them with the `CheckEntityExists` as follows which will only include the ReleaseStatuses for that Release.

```
return await _persistenceHelper
.CheckEntityExists<Release>(releaseId, release =>
  release.Include(r => r.ReleaseStatuses)
         .ThenInclude(rs => rs.CreatedBy))
.OnSuccess(release => {
   ... release.ReleaseStatuses to view model here... 
});
```